dlrover/python/master/watcher/factory.py

```diff
@@ -45,3 +45,21 @@ def new_scale_plan_watcher(platform, job_name, namespace, job_uuid):
         return RayScalePlanWatcher(job_name, namespace, job_uuid)
     else:
         raise ValueError("Not support engine %s", platform)
+
+def new_elasticjob_watcher(args):
+    logger.info("New %s ElasticJob Watcher", args.platform)
+
+    if args.platform in (PlatformType.KUBERNETES, PlatformType.PY_KUBERNETES):
+        from dlrover.python.master.watcher.k8s_watcher import (
+            K8sElasticJobWatcher
+        )
+
+        return K8sElasticJobWatcher(args)
+    elif args.platform in (PlatformType.RAY):
```

Comment:
Remove ray's part. Ray does not need such a watcher(gang scheduling is already done by using placement group).